Lib/idlelib/sidebar.py

```diff
@@ -40,6 +54,15 @@ def get_widget_padding(widget):
     return padx, pady
 
 
+@contextlib.contextmanager
```

Comment:
This or something like it would seem useful in pyshell too.  Yet another spinoff.

Reply:
Indeed!